server/controllers/Profiles.js

```diff
@@ -0,0 +1,73 @@
+import models from '../database/models';
+import { serverResponse, serverError, imageUpload } from '../helpers';
+
+const { User } = models;
+
+/**
+ * @export
+ * @class Profiles
+ */
+class Profiles {
+  /**
+   * @name update
+   * @description allows a user to update their profile
+   * @param {object} req request object
+   * @param {object} res response object
+   * @returns {json} the json response been return by the server
+   * @memberof ProfilesController
+   */
+  static async update(req, res) {
+    try {
+      const {
+        body, user, file, params
+      } = req;
+      const checkId = user.id == params.id;
+      const { email, userName } = body;
+      if (!checkId) {
+        return serverResponse(res, 403, {
+          error: 'you cannot access this route'
+        });
+      }
+      if (email) {
+        const existingEmail = await User.findByEmail(email);
+        if (existingEmail) {
+          return serverResponse(res, 409, {
+            error: 'email has already been taken'
+          });
+        }
+      }
+      if (userName) {
+        const existingUserName = await User.findByUsername(userName);
+        if (existingUserName) {
+          return serverResponse(res, 409, {
+            error: 'username has already been taken'
+          });
+        }
+      }
+      let avatarUrl;
+      if (file) {
+        avatarUrl = await imageUpload(req);
+      }
+      await User.update(
+        {
+          ...body,
```

Comment:
If users can change their emails, then provision should be made to send them verification emails to the new email address they choose

Reply:
Thanks for the feedback but I think that is beyond the scope of the task assigned.